### PR TITLE
Orderbook operations permissions

### PR DESCRIPTION
**Pull request by bragov4ik**
Permissions + tests for them

### CODE REVIEW

File: pallets/order-book/src/lib.rs [hunk 1]
```diff
@@ -453,7 +460,17 @@ pub mod pallet {
             min_lot_size: OrderVolume,
             max_lot_size: OrderVolume,
         ) -> DispatchResult {
-            ensure_root(origin)?;
+            let origin_check_result = T::ParameterUpdateOrigin::ensure_origin(origin)?;
+            match origin_check_result {
+                Either::Left(who) => {
+                    ensure!(
+                        T::AssetInfoProvider::is_asset_owner(&order_book_id.base, &who)
+                            || T::AssetInfoProvider::is_asset_owner(&order_book_id.quote, &who),
```
Comment (wer1st) [inline pallets/order-book/src/lib.rs:468]: Remove. The owner of `base` asset can update order book, not the owner of `quote` asset

File: pallets/order-book/src/lib.rs [hunk 2]
```diff
@@ -551,7 +568,7 @@ pub mod pallet {
             order_book_id: OrderBookId<AssetIdOf<T>>,
             status: OrderBookStatus,
         ) -> DispatchResult {
-            ensure_root(origin)?;
+            let () = T::StatusUpdateOrigin::ensure_origin(origin)?;
```
Comment (wer1st) [inline pallets/order-book/src/lib.rs:571]:
```suggestion
            T::StatusUpdateOrigin::ensure_origin(origin)?;
```

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -81,7 +81,7 @@ use extensions::ChargeTransactionPayment;
 use frame_election_provider_support::{generate_solution_type, onchain, SequentialPhragmen};
 use frame_support::traits::{ConstU128, ConstU32, Currency, EitherOfDiverse};
 use frame_system::offchain::{Account, SigningTypes};
-use frame_system::EnsureRoot;
+use frame_system::{EnsureRoot, EnsureSigned};
```
Comment (wer1st) [inline runtime/src/lib.rs:84]: Move `use frame_system::EnsureSigned` under `wip` feature flag

File: pallets/order-book/src/lib.rs (hunk 1 above)
Comment (bragov4ik) [reply pallets/order-book/src/lib.rs]: https://github.com/sora-xor/sora2-network/issues/400#issuecomment-1571702153

**Review by vovac12 — APPROVED** [review]
Looks great

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -69,6 +69,8 @@ use common::{Description, GetMarketInfo};
 use common::{XOR, XST, XSTUSD};
 use constants::currency::deposit;
 use constants::time::*;
+#[cfg(feature = "wip")]
```
Comment (wer1st) [inline runtime/src/lib.rs:72]:
```suggestion
#[cfg(feature = "wip")] // order-book
```

File: runtime/src/lib.rs [hunk 5]
```diff
@@ -82,6 +84,8 @@ use frame_election_provider_support::{generate_solution_type, onchain, Sequentia
 use frame_support::traits::{ConstU128, ConstU32, Currency, EitherOfDiverse};
 use frame_system::offchain::{Account, SigningTypes};
 use frame_system::EnsureRoot;
+#[cfg(feature = "wip")]
```
Comment (wer1st) [inline runtime/src/lib.rs:87]:
```suggestion
#[cfg(feature = "wip")] // order-book
```

**Review by wer1st — APPROVED** [review]

**Pull request merged**